<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8">
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />



<meta name="progressive" content="false" />
<meta name="allow-skip" content="false" />

<title>Genetic Data Tutorial</title>


<!-- highlightjs -->
<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>

<!-- taken from https://github.com/rstudio/rmarkdown/blob/67b7f5fc779e4cfdfd0f021d3d7745b6b6e17149/inst/rmd/h/default.html#L296-L362 -->
<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<script>
$(document).ready(function () {
  window.buildTabsets("section-TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>
<!-- end tabsets -->



</head>

<body>



<div class="pageContent band">
<div class="bandContent page">

<div class="topics">

<div id="section-learning-objectives" class="section level2">
<h2>Learning Objectives</h2>
<p>This tutorial is about the ways that genetic and epigenetic data are encoded and stored. While we may allude to how this data is analyzed, we will primarily be dealing with file types and data structures. By the end of this section, we expect that you will be able to:</p>
<ol style="list-style-type: decimal">
<li>Identify common file types and what types of data are stored in them</li>
<li>Describe some technical limitations of converting genetic and epigenetic information into computer-friendly formats</li>
<li>Find publicly available data from online repositories and download them to your local computer</li>
</ol>
<p>The evolution of techniques to read DNA will be covered in class next week. We’re introducing the topic now to give you an idea of the complexities and help you contextualize what you’re about to learn.</p>
</div>
<div id="section-representing-the-raw-sequence" class="section level2">
<h2>Representing the raw sequence</h2>
<p>At its simplest, genetic data is an ordered sequence of nucleotides. So, the most obvious way to encode genetic data is to simply write out the nucleotide at each position of the genome. Unsurprisingly, this is how the first geneticists and biochemists recorded DNA sequences.</p>
<p>However, reading the entire genome base by base is hard. The first geneticists and biochemists did not do this. Instead, they focused on specific genes and wrote out the sequences by hand (reading them from a gel, but more on this in class).</p>
<p>After transcripts of genes were widely available, we didn’t transition immediately to sequencing entire chromosomes or genomes. Much of our genetic sequences are conserved, or exactly the same, not just across people or individuals of the same species, but also across species. So for a long time, people only looked for variation. Instead of obtaining a genetic sequence, they assessed which base was present at a position that is known to vary between people.</p>
<p>We didn’t try to sequence the entire genome for a long time because we believed in “junk DNA.” These assumptions, that DNA doesn’t vary much from person to person and that any DNA that isn’t coding DNA is meaningless, have not held up. So nowadays, we do a lot more sequencing and of a lot more things.</p>
<p>Even as we transition to longer sequences of DNA, it is important to remember that we don’t have magnifying glasses powerful enough to read DNA. Instead, we use methods with varying levels of accuracy to decide what we think the sequence of bases is.</p>
<div id="section-fasta" class="section level3">
<h3>FASTA</h3>
<p>One of the first file formats created to store genetic data for use by a computer is called FASTA<a href="#section-fn1" class="footnote-ref" id="section-fnref1"><sup>1</sup></a>. It is very simple.</p>
<p>The below is <a href="https://www.ncbi.nlm.nih.gov/nuccore/NC_000019.10?report=fasta&amp;from=44905796&amp;to=44909393">the reference DNA sequence for APOE</a>, a gene associated with Alzheimer’s disease risk.</p>
<pre style="max-height: 200px;"><code>&gt;NC_000019.10:44905796-44909393 Homo sapiens chromosome 19, GRCh38.p13 Primary Assembly
CTACTCAGCCCCAGCGGAGGTGAAGGACGTCCTTCCCCAGGAGCCGGTGAGAAGCGCAGTCGGGGGCACG
GGGATGAGCTCAGGGGCCTCTAGAAAGAGCTGGGACCCTGGGAACCCCTGGCCTCCAGGTAGTCTCAGGA
GAGCTACTCGGGGTCGGGCTTGGGGAGAGGAGGAGCGGGGGTGAGGCAAGCAGCAGGGGACTGGACCTGG
GAAGGGCTGGGCAGCAGAGACGACCCGACCCGCTAGAAGGTGGGGTGGGGAGAGCAGCTGGACTGGGATG
TAAGCCATAGCAGGACTCCACGAGTTGTCACTATCATTTATCGAGCACCTACTGGGTGTCCCCAGTGTCC
TCAGATCTCCATAACTGGGGAGCCAGGGGCAGCGACACGGTAGCTAGCCGTCGATTGGAGAACTTTAAAA
TGAGGACTGAATTAGCTCATAAATGGAACACGGCGCTTAACTGTGAGGTTGGAGCTTAGAATGTGAAGGG
AGAATGAGGAATGCGAGACTGGGACTGAGATGGAACCGGCGGTGGGGAGGGGGTGGGGGGATGGAATTTG
AACCCCGGGAGAGGAAGATGGAATTTTCTATGGAGGCCGACCTGGGGATGGGGAGATAAGAGAAGACCAG
GAGGGAGTTAAATAGGGAATGGGTTGGGGGCGGCTTGGTAAATGTGCTGGGATTAGGCTGTTGCAGATAA
TGCAACAAGGCTTGGAAGGCTAACCTGGGGTGAGGCCGGGTTGGGGCCGGGCTGGGGGTGGGAGGAGTCC
TCACTGGCGGTTGATTGACAGTTTCTCCTTCCCCAGACTGGCCAATCACAGGCAGGAAGATGAAGGTTCT
GTGGGCTGCGTTGCTGGTCACATTCCTGGCAGGTATGGGGGCGGGGCTTGCTCGGTTCCCCCCGCTCCTC
CCCCTCTCATCCTCACCTCAACCTCCTGGCCCCATTCAGGCAGACCCTGGGCCCCCTCTTCTGAGGCTTC
TGTGCTGCTTCCTGGCTCTGAACAGCGATTTGACGCTCTCTGGGCCTCGGTTTCCCCCATCCTTGAGATA
GGAGTTAGAAGTTGTTTTGTTGTTGTTGTTTGTTGTTGTTGTTTTGTTTTTTTGAGATGAAGTCTCGCTC
TGTCGCCCAGGCTGGAGTGCAGTGGCGGGATCTCGGCTCACTGCAAGCTCCGCCTCCCAGGTCCACGCCA
TTCTCCTGCCTCAGCCTCCCAAGTAGCTGGGACTACAGGCACATGCCACCACACCCGACTAACTTTTTTG
TATTTTCAGTAGAGACGGGGTTTCACCATGTTGGCCAGGCTGGTCTGGAACTCCTGACCTCAGGTGATCT
GCCCGTTTCGATCTCCCAAAGTGCTGGGATTACAGGCGTGAGCCACCGCACCTGGCTGGGAGTTAGAGGT
TTCTAATGCATTGCAGGCAGATAGTGAATACCAGACACGGGGCAGCTGTGATCTTTATTCTCCATCACCC
CCACACAGCCCTGCCTGGGGCACACAAGGACACTCAATACATGCTTTTCCGCTGGGCGCGGTGGCTCACC
CCTGTAATCCCAGCACTTTGGGAGGCCAAGGTGGGAGGATCACTTGAGCCCAGGAGTTCAACACCAGCCT
GGGCAACATAGTGAGACCCTGTCTCTACTAAAAATACAAAAATTAGCCAGGCATGGTGCCACACACCTGT
GCTCTCAGCTACTCAGGAGGCTGAGGCAGGAGGATCGCTTGAGCCCAGAAGGTCAAGGTTGCAGTGAACC
ATGTTCAGGCCGCTGCACTCCAGCCTGGGTGACAGAGCAAGACCCTGTTTATAAATACATAATGCTTTCC
AAGTGATTAAACCGACTCCCCCCTCACCCTGCCCACCATGGCTCCAAAGAAGCATTTGTGGAGCACCTTC
TGTGTGCCCCTAGGTACTAGATGCCTGGACGGGGTCAGAAGGACCCTGACCCACCTTGAACTTGTTCCAC
ACAGGATGCCAGGCCAAGGTGGAGCAAGCGGTGGAGACAGAGCCGGAGCCCGAGCTGCGCCAGCAGACCG
AGTGGCAGAGCGGCCAGCGCTGGGAACTGGCACTGGGTCGCTTTTGGGATTACCTGCGCTGGGTGCAGAC
ACTGTCTGAGCAGGTGCAGGAGGAGCTGCTCAGCTCCCAGGTCACCCAGGAACTGAGGTGAGTGTCCCCA
TCCTGGCCCTTGACCCTCCTGGTGGGCGGCTATACCTCCCCAGGTCCAGGTTTCATTCTGCCCCTGTCGC
TAAGTCTTGGGGGGCCTGGGTCTCTGCTGGTTCTAGCTTCCTCTTCCCATTTCTGACTCCTGGCTTTAGC
TCTCTGGAATTCTCTCTCTCAGCTTTGTCTCTCTCTCTTCCCTTCTGACTCAGTCTCTCACACTCGTCCT
GGCTCTGTCTCTGTCCTTCCCTAGCTCTTTTATATAGAGACAGAGAGATGGGGTCTCACTGTGTTGCCCA
GGCTGGTCTTGAACTTCTGGGCTCAAGCGATCCTCCCGCCTCGGCCTCCCAAAGTGCTGGGATTAGAGGC
ATGAGCCACCTTGCCCGGCCTCCTAGCTCCTTCTTCGTCTCTGCCTCTGCCCTCTGCATCTGCTCTCTGC
ATCTGTCTCTGTCTCCTTCTCTCGGCCTCTGCCCCGTTCCTTCTCTCCCTCTTGGGTCTCTCTGGCTCAT
CCCCATCTCGCCCGCCCCATCCCAGCCCTTCTCCCCGCCTCCCACTGTGCGACACCCTCCCGCCCTCTCG
GCCGCAGGGCGCTGATGGACGAGACCATGAAGGAGTTGAAGGCCTACAAATCGGAACTGGAGGAACAACT
GACCCCGGTGGCGGAGGAGACGCGGGCACGGCTGTCCAAGGAGCTGCAGGCGGCGCAGGCCCGGCTGGGC
GCGGACATGGAGGACGTGTGCGGCCGCCTGGTGCAGTACCGCGGCGAGGTGCAGGCCATGCTCGGCCAGA
GCACCGAGGAGCTGCGGGTGCGCCTCGCCTCCCACCTGCGCAAGCTGCGTAAGCGGCTCCTCCGCGATGC
CGATGACCTGCAGAAGCGCCTGGCAGTGTACCAGGCCGGGGCCCGCGAGGGCGCCGAGCGCGGCCTCAGC
GCCATCCGCGAGCGCCTGGGGCCCCTGGTGGAACAGGGCCGCGTGCGGGCCGCCACTGTGGGCTCCCTGG
CCGGCCAGCCGCTACAGGAGCGGGCCCAGGCCTGGGGCGAGCGGCTGCGCGCGCGGATGGAGGAGATGGG
CAGCCGGACCCGCGACCGCCTGGACGAGGTGAAGGAGCAGGTGGCGGAGGTGCGCGCCAAGCTGGAGGAG
CAGGCCCAGCAGATACGCCTGCAGGCCGAGGCCTTCCAGGCCCGCCTCAAGAGCTGGTTCGAGCCCCTGG
TGGAAGACATGCAGCGCCAGTGGGCCGGGCTGGTGGAGAAGGTGCAGGCTGCCGTGGGCACCAGCGCCGC
CCCTGTGCCCAGCGACAATCACTGAACGCCGAAGCCTGCAGCCATGCGACCCCACGCCACCCCGTGCCTC
CTGCCTCCGCGCAGCCTGCAGCGGGAGACCCTGTCCCCGCCCCAGCCGTCCTCCTGGGGTGGACCCTAGT
TTAATAAAGATTCACCAAGTTTCACGCA</code></pre>
<p>Lines beginning with a “&gt;” are header lines that identify the sequence in the following line. The National Center for Biotechnology Information (NCBI) used the NCBI reference genome position (NC_000019.10:44905796-44909393), the species (Homo sapiens), the chromosome (chromosome 19), and the reference genome version (GRCh38.p13 Primary Assembly) to label this reference sequence from its <a href="https://www.ncbi.nlm.nih.gov/refseq/rsg/">RefSeq database</a>.</p>
<p>The remaining lines represent the nitrogenous bases of the nucleotides at each position of the gene, with the following meanings<a href="#section-fn2" class="footnote-ref" id="section-fnref2"><sup>2</sup></a>:</p>
<table>
<thead>
<tr class="header">
<th align="left">IUB/IUPAC code</th>
<th align="center">Adenine</th>
<th align="center">Cytosine</th>
<th align="center">Guanine</th>
<th align="center">Thymine</th>
<th align="center">Uracil</th>
<th align="left">Mnemonic</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="center">X</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="left"><strong>A</strong>denine</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="center"></td>
<td align="center">X</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="left"><strong>C</strong>ytosine</td>
</tr>
<tr class="odd">
<td align="left">G</td>
<td align="center"></td>
<td align="center"></td>
<td align="center">X</td>
<td align="center"></td>
<td align="center"></td>
<td align="left"><strong>G</strong>uanine</td>
</tr>
<tr class="even">
<td align="left">T</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center">X</td>
<td align="center"></td>
<td align="left"><strong>T</strong>hymine</td>
</tr>
<tr class="odd">
<td align="left">U</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center">X</td>
<td align="left"><strong>U</strong>racil<a href="#section-fn3" class="footnote-ref" id="section-fnref3"><sup>3</sup></a></td>
</tr>
<tr class="even">
<td align="left">Y</td>
<td align="center"></td>
<td align="center">X</td>
<td align="center"></td>
<td align="center">X</td>
<td align="center">X</td>
<td align="left">P<strong>y</strong>rimidine<a href="#section-fn4" class="footnote-ref" id="section-fnref4"><sup>4</sup></a></td>
</tr>
<tr class="odd">
<td align="left">R</td>
<td align="center">X</td>
<td align="center"></td>
<td align="center">X</td>
<td align="center"></td>
<td align="center"></td>
<td align="left">Pu<strong>r</strong>ine<a href="#section-fn5" class="footnote-ref" id="section-fnref5"><sup>5</sup></a></td>
</tr>
<tr class="even">
<td align="left">S</td>
<td align="center"></td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center"></td>
<td align="center"></td>
<td align="left"><strong>S</strong>trong interaction<a href="#section-fn6" class="footnote-ref" id="section-fnref6"><sup>6</sup></a></td>
</tr>
<tr class="odd">
<td align="left">W</td>
<td align="center">X</td>
<td align="center"></td>
<td align="center"></td>
<td align="center">X</td>
<td align="center">X</td>
<td align="left"><strong>W</strong>eak interaction<a href="#section-fn7" class="footnote-ref" id="section-fnref7"><sup>7</sup></a></td>
</tr>
<tr class="even">
<td align="left">K</td>
<td align="center"></td>
<td align="center"></td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="left"><strong>K</strong>etone<a href="#section-fn8" class="footnote-ref" id="section-fnref8"><sup>8</sup></a></td>
</tr>
<tr class="odd">
<td align="left">M</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="left">A<strong>m</strong>ine<a href="#section-fn9" class="footnote-ref" id="section-fnref9"><sup>9</sup></a></td>
</tr>
<tr class="even">
<td align="left">H</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center"></td>
<td align="center">X</td>
<td align="center">X</td>
<td align="left">not G (<strong>H</strong> follows G in the alphabet)</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="center"></td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="left">not A (<strong>B</strong> follows A in the alphabet)</td>
</tr>
<tr class="even">
<td align="left">V</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center"></td>
<td align="center"></td>
<td align="left">not T/U (<strong>V</strong> follows U in the alphabet)</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="center">X</td>
<td align="left">A<strong>n</strong>y</td>
</tr>
<tr class="even">
<td align="left">-</td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="center"></td>
<td align="left">Gap<a href="#section-fn10" class="footnote-ref" id="section-fnref10"><sup>10</sup></a></td>
</tr>
</tbody>
</table>
<p>You can represent mRNA sequences (like this <a href="https://www.ncbi.nlm.nih.gov/nuccore/NM_000041.4">reference for one of APOE’s transcripts</a>) this way too<a href="#section-fn11" class="footnote-ref" id="section-fnref11"><sup>11</sup></a>:</p>
<pre style="max-height: 200px;"><code>&gt;NM_000041.4 Homo sapiens apolipoprotein E (APOE), transcript variant 2, mRNA
CTACTCAGCCCCAGCGGAGGTGAAGGACGTCCTTCCCCAGGAGCCGACTGGCCAATCACAGGCAGGAAGA
TGAAGGTTCTGTGGGCTGCGTTGCTGGTCACATTCCTGGCAGGATGCCAGGCCAAGGTGGAGCAAGCGGT
GGAGACAGAGCCGGAGCCCGAGCTGCGCCAGCAGACCGAGTGGCAGAGCGGCCAGCGCTGGGAACTGGCA
CTGGGTCGCTTTTGGGATTACCTGCGCTGGGTGCAGACACTGTCTGAGCAGGTGCAGGAGGAGCTGCTCA
GCTCCCAGGTCACCCAGGAACTGAGGGCGCTGATGGACGAGACCATGAAGGAGTTGAAGGCCTACAAATC
GGAACTGGAGGAACAACTGACCCCGGTGGCGGAGGAGACGCGGGCACGGCTGTCCAAGGAGCTGCAGGCG
GCGCAGGCCCGGCTGGGCGCGGACATGGAGGACGTGTGCGGCCGCCTGGTGCAGTACCGCGGCGAGGTGC
AGGCCATGCTCGGCCAGAGCACCGAGGAGCTGCGGGTGCGCCTCGCCTCCCACCTGCGCAAGCTGCGTAA
GCGGCTCCTCCGCGATGCCGATGACCTGCAGAAGCGCCTGGCAGTGTACCAGGCCGGGGCCCGCGAGGGC
GCCGAGCGCGGCCTCAGCGCCATCCGCGAGCGCCTGGGGCCCCTGGTGGAACAGGGCCGCGTGCGGGCCG
CCACTGTGGGCTCCCTGGCCGGCCAGCCGCTACAGGAGCGGGCCCAGGCCTGGGGCGAGCGGCTGCGCGC
GCGGATGGAGGAGATGGGCAGCCGGACCCGCGACCGCCTGGACGAGGTGAAGGAGCAGGTGGCGGAGGTG
CGCGCCAAGCTGGAGGAGCAGGCCCAGCAGATACGCCTGCAGGCCGAGGCCTTCCAGGCCCGCCTCAAGA
GCTGGTTCGAGCCCCTGGTGGAAGACATGCAGCGCCAGTGGGCCGGGCTGGTGGAGAAGGTGCAGGCTGC
CGTGGGCACCAGCGCCGCCCCTGTGCCCAGCGACAATCACTGAACGCCGAAGCCTGCAGCCATGCGACCC
CACGCCACCCCGTGCCTCCTGCCTCCGCGCAGCCTGCAGCGGGAGACCCTGTCCCCGCCCCAGCCGTCCT
CCTGGGGTGGACCCTAGTTTAATAAAGATTCACCAAGTTTCACGCA</code></pre>
<p>To represent a protein, simply use the <a href="https://molbiol-tools.ca/Amino_acid_abbreviations.htm">one letter amino acid codes</a> instead of the nucleotide abbreviations.</p>
<table>
<thead>
<tr class="header">
<th align="left">IUB/IUPAC code</th>
<th align="left">Meaning</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">A</td>
<td align="left"><strong>A</strong>lanine</td>
</tr>
<tr class="even">
<td align="left">R</td>
<td align="left">A<strong>r</strong>ginine</td>
</tr>
<tr class="odd">
<td align="left">N</td>
<td align="left">Asparagi<strong>n</strong>e</td>
</tr>
<tr class="even">
<td align="left">D</td>
<td align="left">Aspartate</td>
</tr>
<tr class="odd">
<td align="left">B</td>
<td align="left">Aspartate or Asparagine</td>
</tr>
<tr class="even">
<td align="left">C</td>
<td align="left"><strong>C</strong>ysteine</td>
</tr>
<tr class="odd">
<td align="left">E</td>
<td align="left">Glutamate</td>
</tr>
<tr class="even">
<td align="left">Q</td>
<td align="left">Glutamine</td>
</tr>
<tr class="odd">
<td align="left">Z</td>
<td align="left">Glutamate or Glutamine</td>
</tr>
<tr class="even">
<td align="left">G</td>
<td align="left"><strong>G</strong>lycine</td>
</tr>
<tr class="odd">
<td align="left">H</td>
<td align="left"><strong>H</strong>istidine</td>
</tr>
<tr class="even">
<td align="left">I</td>
<td align="left"><strong>I</strong>soleucine</td>
</tr>
<tr class="odd">
<td align="left">L</td>
<td align="left"><strong>L</strong>eucine</td>
</tr>
<tr class="even">
<td align="left">K</td>
<td align="left">Lysine</td>
</tr>
<tr class="odd">
<td align="left">M</td>
<td align="left"><strong>M</strong>ethionine</td>
</tr>
<tr class="even">
<td align="left">F</td>
<td align="left"><strong>Ph</strong>enylalanine</td>
</tr>
<tr class="odd">
<td align="left">P</td>
<td align="left"><strong>P</strong>roline</td>
</tr>
<tr class="even">
<td align="left">S</td>
<td align="left"><strong>S</strong>erine</td>
</tr>
<tr class="odd">
<td align="left">T</td>
<td align="left"><strong>T</strong>hreonine</td>
</tr>
<tr class="even">
<td align="left">W</td>
<td align="left">Tryptophan</td>
</tr>
<tr class="odd">
<td align="left">Y</td>
<td align="left">T<strong>y</strong>rosine</td>
</tr>
<tr class="even">
<td align="left">V</td>
<td align="left"><strong>V</strong>aline</td>
</tr>
<tr class="odd">
<td align="left">X</td>
<td align="left">Any</td>
</tr>
<tr class="even">
<td align="left">*</td>
<td align="left">Translation stop</td>
</tr>
<tr class="odd">
<td align="left">-</td>
<td align="left">Gap</td>
</tr>
</tbody>
</table>
<p>Here is the <a href="https://www.ncbi.nlm.nih.gov/protein/NP_000032.1">APOE reference amino acid sequence</a>:</p>
<pre><code>&gt;NP_000032.1 apolipoprotein E isoform b precursor [Homo sapiens]
MKVLWAALLVTFLAGCQAKVEQAVETEPEPELRQQTEWQSGQRWELALGRFWDYLRWVQTLSEQVQEELL
SSQVTQELRALMDETMKELKAYKSELEEQLTPVAEETRARLSKELQAAQARLGADMEDVCGRLVQYRGEV
QAMLGQSTEELRVRLASHLRKLRKRLLRDADDLQKRLAVYQAGAREGAERGLSAIRERLGPLVEQGRVRA
ATVGSLAGQPLQERAQAWGERLRARMEEMGSRTRDRLDEVKEQVAEVRAKLEEQAQQIRLQAEAFQARLK
SWFEPLVEDMQRQWAGLVEKVQAAVGTSAAPVPSDNH</code></pre>
<p>You can also have more than one sequence per file, as long as each has its own identification line:</p>
<pre><code>&gt;NP_000032.1 apolipoprotein E isoform b precursor [Homo sapiens]
MKVLWAALLVTFLAGCQAKVEQAVETEPEPELRQQTEWQSGQRWELALGRFWDYLRWVQTLSEQVQEELL
SSQVTQELRALMDETMKELKAYKSELEEQLTPVAEETRARLSKELQAAQARLGADMEDVCGRLVQYRGEV
QAMLGQSTEELRVRLASHLRKLRKRLLRDADDLQKRLAVYQAGAREGAERGLSAIRERLGPLVEQGRVRA
ATVGSLAGQPLQERAQAWGERLRARMEEMGSRTRDRLDEVKEQVAEVRAKLEEQAQQIRLQAEAFQARLK
SWFEPLVEDMQRQWAGLVEKVQAAVGTSAAPVPSDNH
&gt;AXM44625.1 apolipoprotein E, partial [Homo sapiens]
ALMDETMKELKAYKSELEEQLTPVAEETRARLSKELQAAQARLGADMEDVCGRLVQYRGEVQAMLGQSTE
ELRVRLASHLRLLRDADDLQKRLAVYQAGAREGAERGLSAIRERLGPLVEQGRVRAATVGSLAGQPLQER
AQAWGERLRARMEEMGSRTRDRLDEVKEQVAEVRAKLEEQAQQIRLQAEAFQARLKSWFEPLVEDMQRQW
AGLVEKVQAAVGTSAAPVPSDNH</code></pre>
</div>
<div id="section-fastq" class="section level3">
<h3>FASTQ</h3>
<p>The most common way to represent raw sequencing data is with a <a href="https://support.illumina.com/bulletins/2016/04/fastq-files-explained.html">FASTQ file</a>. The sequence for each read generated by the sequencing machine is stored in the same way as in a FASTA file, but with two additional lines encoding the quality of the sequence. The quality scores are where the “Q” in FASTQ comes from.</p>
<p>In two weeks we will be using data from <a href="https://www.internationalgenome.org/about/">the 1000 Genomes project</a> to learn how to use PLINK and determine the global ancestry of samples. I downloaded <a href="https://ftp-trace.ncbi.nih.gov/1000genomes/ftp/sequence_indices/20130502.sequence.index">an index of all available Phase 3 samples</a> and used R to filter the 187,990 samples down to a small number of samples with high coverage to display for you:</p>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":[""],"name":["_rn_"],"type":[""],"align":["left"]},{"label":["FASTQ_FILE"],"name":[1],"type":["chr"],"align":["left"]},{"label":["RUN_ID"],"name":[2],"type":["chr"],"align":["left"]},{"label":["STUDY_ID"],"name":[3],"type":["chr"],"align":["left"]},{"label":["STUDY_NAME"],"name":[4],"type":["chr"],"align":["left"]},{"label":["SUBMISSION_DATE"],"name":[5],"type":["chr"],"align":["left"]},{"label":["POPULATION"],"name":[6],"type":["chr"],"align":["left"]}],"data":[{"1":"data/NA19238/sequence_read/ERR000018.filt.fastq.gz","2":"ERR000018","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"1"},{"1":"data/NA19238/sequence_read/ERR000019.filt.fastq.gz","2":"ERR000019","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"2"},{"1":"data/NA19238/sequence_read/ERR000021.filt.fastq.gz","2":"ERR000021","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"6"},{"1":"data/NA19238/sequence_read/ERR000022.filt.fastq.gz","2":"ERR000022","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"7"},{"1":"data/NA19238/sequence_read/ERR000023.filt.fastq.gz","2":"ERR000023","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"8"},{"1":"data/NA19238/sequence_read/ERR000024.filt.fastq.gz","2":"ERR000024","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"9"},{"1":"data/NA19238/sequence_read/ERR000028.filt.fastq.gz","2":"ERR000028","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"16"},{"1":"data/NA19238/sequence_read/ERR000031.filt.fastq.gz","2":"ERR000031","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"20"},{"1":"data/NA19238/sequence_read/ERR000033.filt.fastq.gz","2":"ERR000033","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"21"},{"1":"data/NA19238/sequence_read/ERR000035.filt.fastq.gz","2":"ERR000035","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"25"},{"1":"data/NA19238/sequence_read/ERR000036.filt.fastq.gz","2":"ERR000036","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"26"},{"1":"data/NA19238/sequence_read/ERR000037.filt.fastq.gz","2":"ERR000037","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"27"},{"1":"data/NA19238/sequence_read/ERR000038.filt.fastq.gz","2":"ERR000038","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"28"},{"1":"data/NA19238/sequence_read/ERR000039.filt.fastq.gz","2":"ERR000039","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"29"},{"1":"data/NA19238/sequence_read/ERR000040.filt.fastq.gz","2":"ERR000040","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"30"},{"1":"data/NA19238/sequence_read/ERR000041.filt.fastq.gz","2":"ERR000041","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"31"},{"1":"data/NA19238/sequence_read/ERR000042.filt.fastq.gz","2":"ERR000042","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-08-14 00:00:00","6":"YRI","_rn_":"32"},{"1":"data/NA19240/sequence_read/ERR001919.filt.fastq.gz","2":"ERR001919","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4590"},{"1":"data/NA19240/sequence_read/ERR001920.filt.fastq.gz","2":"ERR001920","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4591"},{"1":"data/NA19240/sequence_read/ERR001921.filt.fastq.gz","2":"ERR001921","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4592"},{"1":"data/NA19240/sequence_read/ERR001922.filt.fastq.gz","2":"ERR001922","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4593"},{"1":"data/NA19240/sequence_read/ERR001923.filt.fastq.gz","2":"ERR001923","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4594"},{"1":"data/NA19240/sequence_read/ERR001924.filt.fastq.gz","2":"ERR001924","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4595"},{"1":"data/NA19240/sequence_read/ERR001925.filt.fastq.gz","2":"ERR001925","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4596"},{"1":"data/NA19240/sequence_read/ERR001926.filt.fastq.gz","2":"ERR001926","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4597"},{"1":"data/NA19238/sequence_read/ERR001927.filt.fastq.gz","2":"ERR001927","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4598"},{"1":"data/NA19238/sequence_read/ERR001928.filt.fastq.gz","2":"ERR001928","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4599"},{"1":"data/NA19239/sequence_read/ERR001929.filt.fastq.gz","2":"ERR001929","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4600"},{"1":"data/NA19239/sequence_read/ERR001930.filt.fastq.gz","2":"ERR001930","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4601"},{"1":"data/NA19240/sequence_read/ERR001931.filt.fastq.gz","2":"ERR001931","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4602"},{"1":"data/NA19240/sequence_read/ERR001932.filt.fastq.gz","2":"ERR001932","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4603"},{"1":"data/NA19238/sequence_read/ERR001933.filt.fastq.gz","2":"ERR001933","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4604"},{"1":"data/NA19238/sequence_read/ERR001934.filt.fastq.gz","2":"ERR001934","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4605"},{"1":"data/NA19239/sequence_read/ERR001935.filt.fastq.gz","2":"ERR001935","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4606"},{"1":"data/NA19239/sequence_read/ERR001936.filt.fastq.gz","2":"ERR001936","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4607"},{"1":"data/NA19240/sequence_read/ERR001937.filt.fastq.gz","2":"ERR001937","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4608"},{"1":"data/NA19238/sequence_read/ERR001938.filt.fastq.gz","2":"ERR001938","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4609"},{"1":"data/NA19238/sequence_read/ERR001939.filt.fastq.gz","2":"ERR001939","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4610"},{"1":"data/NA19238/sequence_read/ERR001940.filt.fastq.gz","2":"ERR001940","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4611"},{"1":"data/NA19239/sequence_read/ERR001941.filt.fastq.gz","2":"ERR001941","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4612"},{"1":"data/NA19239/sequence_read/ERR001942.filt.fastq.gz","2":"ERR001942","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4613"},{"1":"data/NA19240/sequence_read/ERR001943.filt.fastq.gz","2":"ERR001943","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4614"},{"1":"data/NA19240/sequence_read/ERR001944.filt.fastq.gz","2":"ERR001944","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4615"},{"1":"data/NA19239/sequence_read/ERR001945.filt.fastq.gz","2":"ERR001945","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4616"},{"1":"data/NA19239/sequence_read/ERR001946.filt.fastq.gz","2":"ERR001946","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4617"},{"1":"data/NA19238/sequence_read/ERR001947.filt.fastq.gz","2":"ERR001947","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4618"},{"1":"data/NA19238/sequence_read/ERR001948.filt.fastq.gz","2":"ERR001948","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4619"},{"1":"data/NA19238/sequence_read/ERR001949.filt.fastq.gz","2":"ERR001949","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4620"},{"1":"data/NA19240/sequence_read/ERR001950.filt.fastq.gz","2":"ERR001950","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4621"},{"1":"data/NA19240/sequence_read/ERR001951.filt.fastq.gz","2":"ERR001951","3":"SRP000032","4":"1000Genomes Project Pilot 2","5":"2008-12-15 00:00:00","6":"YRI","_rn_":"4622"}],"options":{"columns":{"min":[5],"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>You can download the first FASTQ on the list to get an idea of what a FASTQ file looks like<a href="#section-fn12" class="footnote-ref" id="section-fnref12"><sup>12</sup></a>:</p>
<pre class="bash"><code>wget ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/NA19238/sequence_read/ERR000018.filt.fastq.gz</code></pre>
<p>We can get all the information about the first read in this file from the first four lines of this file<a href="#section-fn13" class="footnote-ref" id="section-fnref13"><sup>13</sup></a>:</p>
<pre class="bash"><code>gzcat ERR000018.filt.fastq.gz | head -n 4</code></pre>
<pre><code>@ERR000018.1 BGI-FC307N0AAXX_5_1_390:1819
GAATTGAATCGAATAGAAAGTATGTGCATCAATTGG
+
IIIIIIIIIIIIIIIIIIIIIII&gt;IIIIIFIIII+I</code></pre>
<p>Header lines of a FASTQ file begin with the “@” character instead of “&gt;”. In a FASTQ file header lines function as read identifiers and generally contain information like the sample ID, the read group, the flowcell, etc. Different sequencers have <a href="https://en.wikipedia.org/wiki/FASTQ_format#Illumina_sequence_identifiers">specific read identifier formats</a>.</p>
<p>Just as in a FASTA file, the line following the identifier contains the raw sequence. Unlike the FASTA, this line is followed by a separator<a href="#section-fn14" class="footnote-ref" id="section-fnref14"><sup>14</sup></a> (“+”) and a bunch of alphanumeric characters on the fourth line representing the sequence quality.</p>
<p>The most common type of score to be used for this line is called a Phred score. A Phred score is a log transformation of the probability that there was an error for a given base (<span class="math inline">\(q=-10\log_{10}(p)\)</span>). That error probability (<span class="math inline">\(p\)</span>) is based on the surrounding florescence peaks of the sequencing machine. This means that a higher value Phred score represents a higher quality sequence and a lower error probability.</p>
<p><img src="GeneticData_files/figure-html/unnamed-chunk-1-1.png" width="777.6" /></p>
<p>You might notice that these scores are numeric and the quality score line in the FASTQ file is letters and special characters. This is because it takes more memory to store a two digit number like “10” than it does to store a single character. In FASTQ files, the scores are mapped to characters by using a range of the <a href="https://en.wikipedia.org/wiki/ASCII#Printable_characters">ASCII characters</a> in order.</p>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["Phred"],"name":[1],"type":["int"],"align":["right"]},{"label":["ASCII_index"],"name":[2],"type":["int"],"align":["right"]},{"label":["ASCII_character"],"name":[3],"type":["chr"],"align":["left"]}],"data":[{"1":"0","2":"33","3":"!"},{"1":"1","2":"34","3":"\""},{"1":"2","2":"35","3":"#"},{"1":"3","2":"36","3":"$"},{"1":"4","2":"37","3":"%"},{"1":"5","2":"38","3":"&"},{"1":"6","2":"39","3":"'"},{"1":"7","2":"40","3":"("},{"1":"8","2":"41","3":")"},{"1":"9","2":"42","3":"*"},{"1":"10","2":"43","3":"+"},{"1":"11","2":"44","3":","},{"1":"12","2":"45","3":"-"},{"1":"13","2":"46","3":"."},{"1":"14","2":"47","3":"/"},{"1":"15","2":"48","3":"0"},{"1":"16","2":"49","3":"1"},{"1":"17","2":"50","3":"2"},{"1":"18","2":"51","3":"3"},{"1":"19","2":"52","3":"4"},{"1":"20","2":"53","3":"5"},{"1":"21","2":"54","3":"6"},{"1":"22","2":"55","3":"7"},{"1":"23","2":"56","3":"8"},{"1":"24","2":"57","3":"9"},{"1":"25","2":"58","3":":"},{"1":"26","2":"59","3":";"},{"1":"27","2":"60","3":"<"},{"1":"28","2":"61","3":"="},{"1":"29","2":"62","3":">"},{"1":"30","2":"63","3":"?"},{"1":"31","2":"64","3":"@"},{"1":"32","2":"65","3":"A"},{"1":"33","2":"66","3":"B"},{"1":"34","2":"67","3":"C"},{"1":"35","2":"68","3":"D"},{"1":"36","2":"69","3":"E"},{"1":"37","2":"70","3":"F"},{"1":"38","2":"71","3":"G"},{"1":"39","2":"72","3":"H"},{"1":"40","2":"73","3":"I"},{"1":"41","2":"74","3":"J"},{"1":"42","2":"75","3":"K"},{"1":"43","2":"76","3":"L"},{"1":"44","2":"77","3":"M"},{"1":"45","2":"78","3":"N"},{"1":"46","2":"79","3":"O"},{"1":"47","2":"80","3":"P"},{"1":"48","2":"81","3":"Q"},{"1":"49","2":"82","3":"R"},{"1":"50","2":"83","3":"S"},{"1":"51","2":"84","3":"T"},{"1":"52","2":"85","3":"U"},{"1":"53","2":"86","3":"V"},{"1":"54","2":"87","3":"W"},{"1":"55","2":"88","3":"X"},{"1":"56","2":"89","3":"Y"},{"1":"57","2":"90","3":"Z"},{"1":"58","2":"91","3":"["},{"1":"59","2":"92","3":"\\\\"},{"1":"60","2":"93","3":"]"},{"1":"61","2":"94","3":"^"},{"1":"62","2":"95","3":"_"},{"1":"63","2":"96","3":"`"},{"1":"64","2":"97","3":"a"},{"1":"65","2":"98","3":"b"},{"1":"66","2":"99","3":"c"},{"1":"67","2":"100","3":"d"},{"1":"68","2":"101","3":"e"},{"1":"69","2":"102","3":"f"},{"1":"70","2":"103","3":"g"},{"1":"71","2":"104","3":"h"},{"1":"72","2":"105","3":"i"},{"1":"73","2":"106","3":"j"},{"1":"74","2":"107","3":"k"},{"1":"75","2":"108","3":"l"},{"1":"76","2":"109","3":"m"},{"1":"77","2":"110","3":"n"},{"1":"78","2":"111","3":"o"},{"1":"79","2":"112","3":"p"},{"1":"80","2":"113","3":"q"},{"1":"81","2":"114","3":"r"},{"1":"82","2":"115","3":"s"},{"1":"83","2":"116","3":"t"},{"1":"84","2":"117","3":"u"},{"1":"85","2":"118","3":"v"},{"1":"86","2":"119","3":"w"},{"1":"87","2":"120","3":"x"},{"1":"88","2":"121","3":"y"},{"1":"89","2":"122","3":"z"},{"1":"90","2":"123","3":"{"},{"1":"91","2":"124","3":"|"},{"1":"92","2":"125","3":"}"},{"1":"93","2":"126","3":"~"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
<p>This means that the first read from our example FASTQ file has the below Phred scores:</p>
<p><img src="GeneticData_files/figure-html/graph_phred-1.png" width="624" /></p>
<p>This pattern of lower quality base calls toward the end of the read is common.</p>
</div>
</div>
<div id="section-representing-the-aligned-sequence" class="section level2">
<h2>Representing the aligned sequence</h2>
<p>While simply representing the raw sequence with a FASTA or FASTQ file is a good start, most of the time we don’t work with genetic data in that format. In fact, the data that we’re going to use next week to practice variant calling isn’t even available as FASTQ files. This is because the raw reads don’t tell us where in the genome a sequence comes from (or where it <a href="https://genome.sph.umich.edu/w/images/3/3c/Mapping_and_assembly.pdf">maps</a> to). To encode this information, we need a file format capable of encoding the <a href="https://en.wikipedia.org/wiki/Sequence_alignment">alignment</a> of the data.</p>
<div id="section-sam" class="section level3">
<h3>SAM</h3>
<p>The most simple version of this format happens to share my name! <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">SAM (Sequence Alignment/Map)</a> files contain genetic information aligned to a reference genome.</p>
<p>Just like with FASTA and FASTQ files, the header lines of SAM files contain important information about the genetic data within. Unlike raw data files, however, all the header lines in a SAM file must be at the beginning of the file, before any lines representing genetic data. Header lines begin with “@” and have very specific rules about how they should be formatted. We’ll go over the most important ones in a moment.</p>
<p>The alignment of each segment to the reference is represented by a single line with at least 11 tab-delimited fields. To be fully honest, I almost never look at these lines manually because there is little to be learned from them.</p>
<p>If you are consumed by curiosity, here is a simplified version of the <a href="https://galaxyproject.org/tutorials/ngs/sam_fields.png">SAM documentation of the 11 mandatory alignment fields</a>:</p>
<table>
<colgroup>
<col width="11%" />
<col width="13%" />
<col width="29%" />
<col width="26%" />
<col width="17%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Column</th>
<th align="left">Variable</th>
<th align="left">Description</th>
<th align="left">Example</th>
<th align="left">NA value</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">QNAME</td>
<td align="left">Read name</td>
<td align="left">ERR000018.1_BGI-FC307N0AAXX_5_1_390:1819</td>
<td align="left">required</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">FLAG</td>
<td align="left"><a href="https://galaxyproject.org/tutorials/ngs/#flag-field">Bitwise representation</a> of <a href="https://broadinstitute.github.io/picard/explain-flags.html">alignment characteristics</a></td>
<td align="left"></td>
<td align="left">required</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">RNAME</td>
<td align="left">Reference sequence name</td>
<td align="left"></td>
<td align="left">*</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">POS</td>
<td align="left">1-based mapping position of the first base of this read</td>
<td align="left"></td>
<td align="left">0</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">MAPQ</td>
<td align="left">Phred-scaled probability that the mapped position of this read is incorrect.</td>
<td align="left"></td>
<td align="left">255</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="left">CIGAR</td>
<td align="left"><a href="https://galaxyproject.org/tutorials/ngs/#cigar-string">CIGAR string</a> representing actions necessary to map read to this position</td>
<td align="left"></td>
<td align="left">*</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="left">RNEXT</td>
<td align="left">For paired end or split reads, the reference name of the next corresponding read. Set to “=” if both reads mapped to the same chromosome.</td>
<td align="left"></td>
<td align="left">*</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="left">PNEXT</td>
<td align="left">For paired end or split reads, the start position of the next corresponding read.</td>
<td align="left"></td>
<td align="left">0</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">TLEN</td>
<td align="left">For paired end or split reads, the inferred combined length of all reads from this DNA fragment.</td>
<td align="left"></td>
<td align="left">*</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">SEQ</td>
<td align="left">Aligned read sequence on the forward strand.</td>
<td align="left"></td>
<td align="left">*</td>
</tr>
<tr class="odd">
<td align="left">11</td>
<td align="left">QUAL</td>
<td align="left">ASCII of Phred-scaled base quality scores (same as in the FASTQ).</td>
<td align="left"></td>
<td align="left">*</td>
</tr>
</tbody>
</table>
</div>
<div id="section-bam" class="section level3">
<h3>BAM</h3>
<p>A BAM is a binary SAM file, this means that the information from the SAM file as described above has been <a href="">compressed</a> so that the file takes up less room. This is incredibly important, because high coverage whole-genome BAM files generally take up more than 100GB even with compression. In fact, there is an additional file format with even more compression called <a href="http://samtools.github.io/hts-specs/CRAMv3.pdf">CRAM</a>. Both BAM and CRAM files generally require index files (BAI and CRAI files) to enable retrieving alignments for a portion of the genome without traversing the whole alignment.</p>
<p>The BAM file with the <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/phase3/data/NA19238/high_coverage_alignment/">mapped reads from the 1000G FASTQ we previously looked at</a> has very high coverage and is 229GB! To download the file to Hoffman2 took several hours and I had to save it to temporary storage (/u/scratch/s/samlj/week_1_GeneticData).</p>
<p>You should be able to view the file’s header lines using <a href="http://www.htslib.org/">Samtools</a>:</p>
<pre class="bash"><code>module load samtools
samtools -H /u/scratch/s/samlj/week_1_GeneticData/NA19238.mapped.ILLUMINA.bwa.YRI.high_coverage_pcr_free.20130924.bam</code></pre>
<pre style="max-height: 200px;"><code>@HD VN:1.4  GO:none SO:coordinate
@SQ SN:1    LN:249250621
@SQ SN:2    LN:243199373
@SQ SN:3    LN:198022430
@SQ SN:4    LN:191154276
@SQ SN:5    LN:180915260
@SQ SN:6    LN:171115067
@SQ SN:7    LN:159138663
@SQ SN:8    LN:146364022
@SQ SN:9    LN:141213431
@SQ SN:10   LN:135534747
@SQ SN:11   LN:135006516
@SQ SN:12   LN:133851895
@SQ SN:13   LN:115169878
@SQ SN:14   LN:107349540
@SQ SN:15   LN:102531392
@SQ SN:16   LN:90354753
@SQ SN:17   LN:81195210
@SQ SN:18   LN:78077248
@SQ SN:19   LN:59128983
@SQ SN:20   LN:63025520
@SQ SN:21   LN:48129895
@SQ SN:22   LN:51304566
@SQ SN:X    LN:155270560
@SQ SN:Y    LN:59373566
@SQ SN:MT   LN:16569
@SQ SN:GL000207.1   LN:4262
@SQ SN:GL000226.1   LN:15008
@SQ SN:GL000229.1   LN:19913
@SQ SN:GL000231.1   LN:27386
@SQ SN:GL000210.1   LN:27682
@SQ SN:GL000239.1   LN:33824
@SQ SN:GL000235.1   LN:34474
@SQ SN:GL000201.1   LN:36148
@SQ SN:GL000247.1   LN:36422
@SQ SN:GL000245.1   LN:36651
@SQ SN:GL000197.1   LN:37175
@SQ SN:GL000203.1   LN:37498
@SQ SN:GL000246.1   LN:38154
@SQ SN:GL000249.1   LN:38502
@SQ SN:GL000196.1   LN:38914
@SQ SN:GL000248.1   LN:39786
@SQ SN:GL000244.1   LN:39929
@SQ SN:GL000238.1   LN:39939
@SQ SN:GL000202.1   LN:40103
@SQ SN:GL000234.1   LN:40531
@SQ SN:GL000232.1   LN:40652
@SQ SN:GL000206.1   LN:41001
@SQ SN:GL000240.1   LN:41933
@SQ SN:GL000236.1   LN:41934
@SQ SN:GL000241.1   LN:42152
@SQ SN:GL000243.1   LN:43341
@SQ SN:GL000242.1   LN:43523
@SQ SN:GL000230.1   LN:43691
@SQ SN:GL000237.1   LN:45867
@SQ SN:GL000233.1   LN:45941
@SQ SN:GL000204.1   LN:81310
@SQ SN:GL000198.1   LN:90085
@SQ SN:GL000208.1   LN:92689
@SQ SN:GL000191.1   LN:106433
@SQ SN:GL000227.1   LN:128374
@SQ SN:GL000228.1   LN:129120
@SQ SN:GL000214.1   LN:137718
@SQ SN:GL000221.1   LN:155397
@SQ SN:GL000209.1   LN:159169
@SQ SN:GL000218.1   LN:161147
@SQ SN:GL000220.1   LN:161802
@SQ SN:GL000213.1   LN:164239
@SQ SN:GL000211.1   LN:166566
@SQ SN:GL000199.1   LN:169874
@SQ SN:GL000217.1   LN:172149
@SQ SN:GL000216.1   LN:172294
@SQ SN:GL000215.1   LN:172545
@SQ SN:GL000205.1   LN:174588
@SQ SN:GL000219.1   LN:179198
@SQ SN:GL000224.1   LN:179693
@SQ SN:GL000223.1   LN:180455
@SQ SN:GL000195.1   LN:182896
@SQ SN:GL000212.1   LN:186858
@SQ SN:GL000222.1   LN:186861
@SQ SN:GL000200.1   LN:187035
@SQ SN:GL000193.1   LN:189789
@SQ SN:GL000194.1   LN:191469
@SQ SN:GL000225.1   LN:211173
@SQ SN:GL000192.1   LN:547496
@SQ SN:NC_007605    LN:171823
@SQ SN:hs37d5   LN:35477943
@RG ID:NA19238  PL:illumina PU:illumina SM:NA19238  CN:illumina
@PG ID:GATK IndelRealigner  VN:2.7-4-gb4c75f6   CL:knownAlleles=[(RodBinding name=knownAlleles source=/humgen/gsa-hpprojects/GATK/bundle/current/b37/dbsnp_137.b37.vcf)] targetIntervals=/humgen/gsa-firehose2/carneiro/yri/NA19238/NA19238.intervals LODThresholdForCleaning=5.0 consensusDeterminationModel=USE_READS entropyThreshold=0.15 maxReadsInMemory=150000 maxIsizeForMovement=3000 maxPositionalMoveAllowed=200 maxConsensuses=30 maxReadsForConsensuses=120 maxReadsForRealignment=20000 noOriginalAlignmentTags=false nWayOut=null generate_nWayOut_md5s=false check_early=false noPGTag=false keepPGTags=false indelsFileForDebugging=null statisticsFileForDebugging=null SNPsFileForDebugging=null
@PG ID:GATK PrintReads  VN:2.7-4-gb4c75f6   CL:readGroup=null platform=null number=-1 sample_file=[] sample_name=[] simplify=false no_pg_tag=false</code></pre>
<p>This BAM file has relatively few header lines in it. BAMs that I’ve generated while doing NGS data processing generally have hundreds of header lines. I’ll explain what the header lines of this file mean, but you’ll likely have to consult <a href="https://samtools.github.io/hts-specs/SAMv1.pdf">the SAM file format reference</a> for other questions. I have spent more of my life reading that document than I ever could have imagined.</p>
<p>As I mentioned previously, SAM format header lines begin with “@”. Immediately following that character is a tag - a two letter abbreviation that tells the computer what information that line contains. Here are the tags found in the 1000G BAM’s header lines:</p>
<table>
<colgroup>
<col width="6%" />
<col width="48%" />
<col width="33%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">Tag</th>
<th align="left">Line contains</th>
<th align="left">Additional rules</th>
<th align="left">Example from file</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>@HD</code></td>
<td align="left">File level metadata, including the SAM format version (<code>VN</code>). May also include how the file is sorted (<code>SO</code>) and the grouping of aligned reads (<code>GO</code>).</td>
<td align="left">There can only be one <code>@HD</code> line and it must be the first line of the file if it exists.</td>
<td align="left"><code>@HD  VN:1.4  GO:none SO:coordinate</code></td>
</tr>
<tr class="even">
<td align="left"><code>@SQ</code></td>
<td align="left">Reference sequence definitions, including the sequence name (<code>SN</code>) and reference sequence length (<code>LN</code>).</td>
<td align="left">The order of these header lines will determine the sort order of the alignment lines when the file is <code>coordinate</code> sorted as this example BAM is. For this reason, all <code>SN</code> tags must be unique.</td>
<td align="left"><code>@SQ    SN:1    LN:249250621</code></td>
</tr>
<tr class="odd">
<td align="left"><code>@RG</code></td>
<td align="left">Read group information, including the read group identifier (<code>ID</code>). May also include the center the sequencing was carried out at (<code>CN</code>), the platform or technology used to produce the reads (<code>PL</code>) the flowcell or slide identifier (<code>PU</code>), and the sample or pool identifier (<code>SM</code>).</td>
<td align="left"><code>@RG</code> lines do not need to be ordered.</td>
<td align="left"><code>@RG  ID:NA19238  PL:illumina PU:illumina SM:NA19238  CN:illumina</code></td>
</tr>
<tr class="even">
<td align="left"><code>@PG</code></td>
<td align="left">Details of a program used to process these reads, including a unique identifier for the program run (<code>ID</code>). May also include the command-line call used to run this program (<code>CL</code>) and the software version (<code>VN</code>).</td>
<td align="left"><code>@PG</code> identifiers must be unique. If the same program was called more than once or there is a conflict because of a merge, the IDs will have to be modified.</td>
<td align="left"><code>@PG ID:GATK PrintReads  VN:2.7-4-gb4c75f6   CL:readGroup=null ...</code></td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="section-representing-the-variants" class="section level2">
<h2>Representing the variants</h2>
<p>FASTQs and BAMs generally contain all of the genetic data for an individual. As we mentioned, this makes them large files that are difficult to process and store. It also means that it is easier to identify individuals in those files than it is if we simply have a file with the genotypes of an individual at variant sites. For this reason, most open access genetic data is in VCF or PLINK file format.</p>
<p>Both file formats are also common ways to represent genotype data from microarray chips.</p>
<div id="section-vcf" class="section level3">
<h3>VCF</h3>
<p><a href="http://samtools.github.io/hts-specs/VCFv4.3.pdf">VCF (Variant Call Format)</a> files are the most common way to represent genotypes after variant calling. The VCF files for the 1000G data we’ve been looking at can be found on <a href="ftp://ftp.1000genomes.ebi.ac.uk/vol1/ftp/release/20130502/">the FTP site</a>.</p>
<p>These files also take a long time to download, although they are significantly smaller than BAM files, so you can find the downloaded files in Sam’s class directory (/u/home/class/hg236/hg236-03/week_3_PLINK/1000G_VCFs).</p>
<p>Like BAM files, VCF files too have a large number of header lines at the beginning of the file that describe how the variant calls were made and determine the organization of the fields. They begin with “#”. Using <a href="http://www.htslib.org/doc/bcftools.html">bcftools</a>, the analogue of samtools for VCFs and BCFs (binary VCFs), you can view just the header lines of one of the VCFs on Hoffman2:</p>
<pre class="bash"><code>module load bcftools
bcftools view -h /u/home/class/hg236/hg236-03/week_3_PLINK/1000G_VCFs/ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz</code></pre>
<pre style="max-height: 200px;"><code>##fileformat=VCFv4.1
##FILTER=&lt;ID=PASS,Description=&quot;All filters passed&quot;&gt;
##fileDate=20150218
##reference=ftp://ftp.1000genomes.ebi.ac.uk//vol1/ftp/technical/reference/phase2_reference_assembly_sequence/hs37d5.fa.gz
##contig=&lt;ID=1,length=249250621,assembly=b37&gt;
##contig=&lt;ID=2,length=243199373,assembly=b37&gt;
##contig=&lt;ID=3,length=198022430,assembly=b37&gt;
##contig=&lt;ID=4,length=191154276,assembly=b37&gt;
##contig=&lt;ID=5,length=180915260,assembly=b37&gt;
##contig=&lt;ID=6,length=171115067,assembly=b37&gt;
##contig=&lt;ID=7,length=159138663,assembly=b37&gt;
##contig=&lt;ID=8,length=146364022,assembly=b37&gt;
##contig=&lt;ID=9,length=141213431,assembly=b37&gt;
##contig=&lt;ID=10,length=135534747,assembly=b37&gt;
##contig=&lt;ID=11,length=135006516,assembly=b37&gt;
##contig=&lt;ID=12,length=133851895,assembly=b37&gt;
##contig=&lt;ID=13,length=115169878,assembly=b37&gt;
##contig=&lt;ID=14,length=107349540,assembly=b37&gt;
##contig=&lt;ID=15,length=102531392,assembly=b37&gt;
##contig=&lt;ID=16,length=90354753,assembly=b37&gt;
##contig=&lt;ID=17,length=81195210,assembly=b37&gt;
##contig=&lt;ID=18,length=78077248,assembly=b37&gt;
##contig=&lt;ID=19,length=59128983,assembly=b37&gt;
##contig=&lt;ID=20,length=63025520,assembly=b37&gt;
##contig=&lt;ID=21,length=48129895,assembly=b37&gt;
##contig=&lt;ID=22,length=51304566,assembly=b37&gt;
##contig=&lt;ID=X,length=155270560,assembly=b37&gt;
##contig=&lt;ID=Y,length=59373566,assembly=b37&gt;
##contig=&lt;ID=MT,length=16569,assembly=b37&gt;
##contig=&lt;ID=GL000207.1,length=4262,assembly=b37&gt;
##contig=&lt;ID=GL000226.1,length=15008,assembly=b37&gt;
##contig=&lt;ID=GL000229.1,length=19913,assembly=b37&gt;
##contig=&lt;ID=GL000231.1,length=27386,assembly=b37&gt;
##contig=&lt;ID=GL000210.1,length=27682,assembly=b37&gt;
##contig=&lt;ID=GL000239.1,length=33824,assembly=b37&gt;
##contig=&lt;ID=GL000235.1,length=34474,assembly=b37&gt;
##contig=&lt;ID=GL000201.1,length=36148,assembly=b37&gt;
##contig=&lt;ID=GL000247.1,length=36422,assembly=b37&gt;
##contig=&lt;ID=GL000245.1,length=36651,assembly=b37&gt;
##contig=&lt;ID=GL000197.1,length=37175,assembly=b37&gt;
##contig=&lt;ID=GL000203.1,length=37498,assembly=b37&gt;
##contig=&lt;ID=GL000246.1,length=38154,assembly=b37&gt;
##contig=&lt;ID=GL000249.1,length=38502,assembly=b37&gt;
##contig=&lt;ID=GL000196.1,length=38914,assembly=b37&gt;
##contig=&lt;ID=GL000248.1,length=39786,assembly=b37&gt;
##contig=&lt;ID=GL000244.1,length=39929,assembly=b37&gt;
##contig=&lt;ID=GL000238.1,length=39939,assembly=b37&gt;
##contig=&lt;ID=GL000202.1,length=40103,assembly=b37&gt;
##contig=&lt;ID=GL000234.1,length=40531,assembly=b37&gt;
##contig=&lt;ID=GL000232.1,length=40652,assembly=b37&gt;
##contig=&lt;ID=GL000206.1,length=41001,assembly=b37&gt;
##contig=&lt;ID=GL000240.1,length=41933,assembly=b37&gt;
##contig=&lt;ID=GL000236.1,length=41934,assembly=b37&gt;
##contig=&lt;ID=GL000241.1,length=42152,assembly=b37&gt;
##contig=&lt;ID=GL000243.1,length=43341,assembly=b37&gt;
##contig=&lt;ID=GL000242.1,length=43523,assembly=b37&gt;
##contig=&lt;ID=GL000230.1,length=43691,assembly=b37&gt;
##contig=&lt;ID=GL000237.1,length=45867,assembly=b37&gt;
##contig=&lt;ID=GL000233.1,length=45941,assembly=b37&gt;
##contig=&lt;ID=GL000204.1,length=81310,assembly=b37&gt;
##contig=&lt;ID=GL000198.1,length=90085,assembly=b37&gt;
##contig=&lt;ID=GL000208.1,length=92689,assembly=b37&gt;
##contig=&lt;ID=GL000191.1,length=106433,assembly=b37&gt;
##contig=&lt;ID=GL000227.1,length=128374,assembly=b37&gt;
##contig=&lt;ID=GL000228.1,length=129120,assembly=b37&gt;
##contig=&lt;ID=GL000214.1,length=137718,assembly=b37&gt;
##contig=&lt;ID=GL000221.1,length=155397,assembly=b37&gt;
##contig=&lt;ID=GL000209.1,length=159169,assembly=b37&gt;
##contig=&lt;ID=GL000218.1,length=161147,assembly=b37&gt;
##contig=&lt;ID=GL000220.1,length=161802,assembly=b37&gt;
##contig=&lt;ID=GL000213.1,length=164239,assembly=b37&gt;
##contig=&lt;ID=GL000211.1,length=166566,assembly=b37&gt;
##contig=&lt;ID=GL000199.1,length=169874,assembly=b37&gt;
##contig=&lt;ID=GL000217.1,length=172149,assembly=b37&gt;
##contig=&lt;ID=GL000216.1,length=172294,assembly=b37&gt;
##contig=&lt;ID=GL000215.1,length=172545,assembly=b37&gt;
##contig=&lt;ID=GL000205.1,length=174588,assembly=b37&gt;
##contig=&lt;ID=GL000219.1,length=179198,assembly=b37&gt;
##contig=&lt;ID=GL000224.1,length=179693,assembly=b37&gt;
##contig=&lt;ID=GL000223.1,length=180455,assembly=b37&gt;
##contig=&lt;ID=GL000195.1,length=182896,assembly=b37&gt;
##contig=&lt;ID=GL000212.1,length=186858,assembly=b37&gt;
##contig=&lt;ID=GL000222.1,length=186861,assembly=b37&gt;
##contig=&lt;ID=GL000200.1,length=187035,assembly=b37&gt;
##contig=&lt;ID=GL000193.1,length=189789,assembly=b37&gt;
##contig=&lt;ID=GL000194.1,length=191469,assembly=b37&gt;
##contig=&lt;ID=GL000225.1,length=211173,assembly=b37&gt;
##contig=&lt;ID=GL000192.1,length=547496,assembly=b37&gt;
##contig=&lt;ID=NC_007605,length=171823,assembly=b37&gt;
##contig=&lt;ID=hs37d5,length=35477943,assembly=b37&gt;
##source=freeBayes v0.9.9.2 | GT values over-written with maximum likelihood state (subject to threshold) OR phylogenetic imputation
##INFO=&lt;ID=DP,Number=1,Type=Integer,Description=&quot;Total read depth at the locus&quot;&gt;
##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;
##source=GenomeSTRiP_v1.04
##ALT=&lt;ID=CNV,Description=&quot;Copy number polymorphism&quot;&gt;
##FILTER=&lt;ID=ALIGNLENGTH,Description=&quot;GSELENGTH &lt; 200&quot;&gt;
##FILTER=&lt;ID=CLUSTERSEP,Description=&quot;GSCLUSTERSEP == NA || GSCLUSTERSEP &lt;= 2.0&quot;&gt;
##FILTER=&lt;ID=DUPLICATE,Description=&quot;GSDUPLICATESCORE != NA &amp;&amp; GSDUPLICATEOVERLAP &gt;= 0.5 &amp;&amp; GSDUPLICATESCORE &gt;= 0.0&quot;&gt;
##FILTER=&lt;ID=GTDEPTH,Description=&quot;GSM1 == NA || GSM1 &lt;= 0.5 || GSM1 &gt;= 2.0&quot;&gt;
##FILTER=&lt;ID=INBREEDINGCOEFF,Description=&quot;GLINBREEDINGCOEFF != NA &amp;&amp; GLINBREEDINGCOEFF &lt; -0.15&quot;&gt;
##FILTER=&lt;ID=NONVARIANT,Description=&quot;GSNONVARSCORE != NA &amp;&amp; GSNONVARSCORE &gt;= 13.0&quot;&gt;
##FORMAT=&lt;ID=CN,Number=1,Type=Integer,Description=&quot;Copy number genotype for imprecise events&quot;&gt;
##FORMAT=&lt;ID=CNL,Number=.,Type=Float,Description=&quot;Copy number likelihoods with no frequency prior&quot;&gt;
##FORMAT=&lt;ID=CNP,Number=.,Type=Float,Description=&quot;Copy number likelihoods&quot;&gt;
##FORMAT=&lt;ID=CNQ,Number=1,Type=Float,Description=&quot;Copy number genotype quality for imprecise events&quot;&gt;
##FORMAT=&lt;ID=GP,Number=G,Type=Float,Description=&quot;Genotype likelihoods&quot;&gt;
##FORMAT=&lt;ID=GQ,Number=1,Type=Integer,Description=&quot;Genotype Quality&quot;&gt;
##FORMAT=&lt;ID=FT,Number=1,Type=String,Description=&quot;Per-sample genotype filter&quot;&gt;
##FORMAT=&lt;ID=PL,Number=G,Type=Integer,Description=&quot;Normalized, Phred-scaled likelihoods for genotypes as defined in the VCF specification&quot;&gt;
##INFO=&lt;ID=END,Number=1,Type=Integer,Description=&quot;End coordinate of this variant&quot;&gt;
##INFO=&lt;ID=SVTYPE,Number=1,Type=String,Description=&quot;Type of structural variant&quot;&gt;
##INFO=&lt;ID=AA,Number=1,Type=String,Description=&quot;Ancestral allele&quot;&gt;
##INFO=&lt;ID=AC,Number=A,Type=Integer,Description=&quot;Total number of alternate alleles in called genotypes&quot;&gt;
##INFO=&lt;ID=AF,Number=A,Type=Float,Description=&quot;Estimated allele frequency in the range (0,1]&quot;&gt;
##INFO=&lt;ID=NS,Number=1,Type=Integer,Description=&quot;Number of samples with data&quot;&gt;
##INFO=&lt;ID=AN,Number=1,Type=Integer,Description=&quot;Total number of alleles in called genotypes&quot;&gt;
##INFO=&lt;ID=SAS_AF,Number=A,Type=Float,Description=&quot;Allele frequency in the SAS populations calculated from AC and AN, in the range (0,1)&quot;&gt;
##INFO=&lt;ID=EUR_AF,Number=A,Type=Float,Description=&quot;Allele frequency in the EUR populations calculated from AC and AN, in the range (0,1)&quot;&gt;
##INFO=&lt;ID=AFR_AF,Number=A,Type=Float,Description=&quot;Allele frequency in the AFR populations calculated from AC and AN, in the range (0,1)&quot;&gt;
##INFO=&lt;ID=AMR_AF,Number=A,Type=Float,Description=&quot;Allele frequency in the AMR populations calculated from AC and AN, in the range (0,1)&quot;&gt;
##INFO=&lt;ID=EAS_AF,Number=A,Type=Float,Description=&quot;Allele frequency in the EAS populations calculated from AC and AN, in the range (0,1)&quot;&gt;
##INFO=&lt;ID=VT,Number=.,Type=String,Description=&quot;indicates what type of variant the line represents&quot;&gt;
##INFO=&lt;ID=EX_TARGET,Number=0,Type=Flag,Description=&quot;indicates whether a variant is within the exon pull down target boundaries&quot;&gt;
##INFO=&lt;ID=MULTI_ALLELIC,Number=0,Type=Flag,Description=&quot;indicates whether a site is multi-allelic&quot;&gt;
##bcftools_viewVersion=1.10.2+htslib-1.10.2
##bcftools_viewCommand=view -h ALL.chrY.phase3_integrated_v2a.20130502.genotypes.vcf.gz; Date=Thu Oct  8 23:28:55 2020
#CHROM  POS ID  REF ALT QUAL    FILTER  INFO    FORMAT  HG00096 HG00101 HG00103 HG00105 HG00107 HG00108 HG00109 HG00112 HG00113 HG00114 HG00115 HG00116 HG00117 HG00119 HG00126 HG00129 HG00131 HG00136 HG00138 HG00139 HG00140 HG00141 HG00142 HG00143 HG00145 HG00148 HG00149 HG00151 HG00155 HG00157 HG00159 HG00160 HG00181 HG00182 HG00183 HG00185 HG00186 HG00187 HG00188 HG00189 HG00190 HG00234 HG00242 HG00243 HG00244 HG00246 HG00251 HG00252 HG00256 HG00260 HG00264 HG00265 HG00267 HG00271 HG00273 HG00277 HG00278 HG00280 HG00284 HG00290 HG00308 HG00310 HG00311 HG00321 HG00325 HG00329 HG00335 HG00336 HG00338 HG00341 HG00342 HG00345 HG00351 HG00358 HG00360 HG00366 HG00369 HG00371 HG00372 HG00375 HG00382 HG00403 HG00406 HG00409 HG00421 HG00436 HG00442 HG00445 HG00448 HG00451 HG00457 HG00463 HG00472 HG00475 HG00478 HG00500 HG00524 HG00530 HG00533 HG00536 HG00542 HG00553 HG00556 HG00559 HG00565 HG00580 HG00583 HG00589 HG00592 HG00595 HG00598 HG00607 HG00610 HG00613 HG00619 HG00622 HG00625 HG00628 HG00631 HG00634 HG00637 HG00640 HG00650 HG00653 HG00656 HG00662 HG00671 HG00674 HG00683 HG00689 HG00692 HG00698 HG00701 HG00704 HG00707 HG00728 HG00731 HG00736 HG00739 HG00742 HG00844 HG00881 HG00982 HG01028 HG01031 HG01047 HG01048 HG01051 HG01054 HG01060 HG01063 HG01066 HG01069 HG01072 HG01075 HG01079 HG01082 HG01085 HG01088 HG01094 HG01097 HG01101 HG01104 HG01107 HG01110 HG01112 HG01121 HG01124 HG01130 HG01133 HG01136 HG01139 HG01142 HG01148 HG01161 HG01164 HG01167 HG01170 HG01173 HG01176 HG01182 HG01187 HG01190 HG01197 HG01200 HG01204 HG01241 HG01247 HG01250 HG01253 HG01256 HG01259 HG01271 HG01277 HG01280 HG01286 HG01302 HG01305 HG01308 HG01311 HG01325 HG01334 HG01341 HG01344 HG01350 HG01353 HG01356 HG01359 HG01362 HG01365 HG01374 HG01377 HG01383 HG01389 HG01392 HG01395 HG01398 HG01402 HG01405 HG01412 HG01413 HG01431 HG01437 HG01440 HG01443 HG01455 HG01461 HG01464 HG01479 HG01485 HG01488 HG01491 HG01494 HG01497 HG01500 HG01503 HG01506 HG01509 HG01512 HG01515 HG01518 HG01521 HG01524 HG01527 HG01530 HG01536 HG01550 HG01556 HG01565 HG01571 HG01577 HG01583 HG01586 HG01589 HG01596 HG01603 HG01606 HG01608 HG01610 HG01615 HG01617 HG01619 HG01624 HG01625 HG01630 HG01631 HG01669 HG01672 HG01675 HG01678 HG01680 HG01682 HG01686 HG01694 HG01699 HG01700 HG01705 HG01708 HG01709 HG01747 HG01756 HG01761 HG01765 HG01767 HG01771 HG01775 HG01777 HG01781 HG01783 HG01785 HG01789 HG01791 HG01810 HG01811 HG01816 HG01840 HG01842 HG01844 HG01846 HG01849 HG01852 HG01860 HG01861 HG01864 HG01865 HG01866 HG01867 HG01872 HG01873 HG01879 HG01882 HG01885 HG01890 HG01892 HG01912 HG01914 HG01917 HG01920 HG01923 HG01926 HG01932 HG01935 HG01938 HG01941 HG01944 HG01947 HG01950 HG01953 HG01961 HG01967 HG01970 HG01974 HG01977 HG01979 HG01982 HG01986 HG01988 HG01990 HG01991 HG02002 HG02008 HG02009 HG02013 HG02014 HG02017 HG02020 HG02023 HG02026 HG02029 HG02032 HG02035 HG02040 HG02047 HG02050 HG02051 HG02053 HG02058 HG02061 HG02064 HG02067 HG02070 HG02073 HG02076 HG02079 HG02082 HG02085 HG02088 HG02090 HG02104 HG02107 HG02116 HG02122 HG02128 HG02131 HG02134 HG02137 HG02138 HG02141 HG02143 HG02146 HG02150 HG02219 HG02221 HG02224 HG02231 HG02233 HG02236 HG02238 HG02250 HG02253 HG02255 HG02259 HG02262 HG02265 HG02271 HG02274 HG02277 HG02281 HG02283 HG02284 HG02285 HG02291 HG02299 HG02304 HG02307 HG02314 HG02317 HG02323 HG02330 HG02332 HG02334 HG02343 HG02351 HG02353 HG02355 HG02356 HG02360 HG02364 HG02367 HG02371 HG02373 HG02374 HG02375 HG02379 HG02380 HG02382 HG02383 HG02384 HG02385 HG02386 HG02389 HG02390 HG02391 HG02392 HG02394 HG02395 HG02396 HG02397 HG02398 HG02399 HG02401 HG02402 HG02406 HG02407 HG02408 HG02409 HG02410 HG02420 HG02429 HG02433 HG02439 HG02442 HG02445 HG02449 HG02455 HG02461 HG02464 HG02470 HG02481 HG02484 HG02489 HG02490 HG02493 HG02496 HG02501 HG02512 HG02521 HG02536 HG02541 HG02545 HG02554 HG02557 HG02561 HG02570 HG02573 HG02582 HG02585 HG02588 HG02594 HG02597 HG02600 HG02603 HG02610 HG02613 HG02620 HG02623 HG02628 HG02634 HG02642 HG02645 HG02648 HG02651 HG02654 HG02657 HG02660 HG02666 HG02675 HG02678 HG02681 HG02684 HG02687 HG02690 HG02696 HG02699 HG02702 HG02715 HG02721 HG02724 HG02727 HG02733 HG02736 HG02756 HG02759 HG02768 HG02771 HG02774 HG02780 HG02783 HG02786 HG02789 HG02792 HG02798 HG02804 HG02807 HG02810 HG02813 HG02816 HG02819 HG02836 HG02839 HG02851 HG02854 HG02860 HG02878 HG02881 HG02884 HG02887 HG02890 HG02895 HG02923 HG02938 HG02941 HG02944 HG02947 HG02953 HG02968 HG02971 HG02973 HG02977 HG02981 HG02982 HG03006 HG03009 HG03012 HG03015 HG03018 HG03021 HG03024 HG03027 HG03039 HG03045 HG03048 HG03054 HG03057 HG03060 HG03063 HG03066 HG03069 HG03072 HG03074 HG03077 HG03078 HG03081 HG03084 HG03096 HG03100 HG03103 HG03109 HG03112 HG03115 HG03118 HG03120 HG03124 HG03127 HG03130 HG03133 HG03136 HG03139 HG03157 HG03160 HG03163 HG03166 HG03169 HG03172 HG03175 HG03190 HG03193 HG03196 HG03199 HG03202 HG03209 HG03224 HG03225 HG03228 HG03234 HG03237 HG03240 HG03246 HG03258 HG03265 HG03268 HG03271 HG03280 HG03295 HG03298 HG03301 HG03304 HG03311 HG03313 HG03343 HG03352 HG03367 HG03370 HG03376 HG03382 HG03385 HG03388 HG03391 HG03394 HG03397 HG03432 HG03433 HG03436 HG03439 HG03442 HG03445 HG03451 HG03457 HG03460 HG03469 HG03472 HG03478 HG03484 HG03490 HG03515 HG03518 HG03521 HG03538 HG03547 HG03556 HG03559 HG03565 HG03571 HG03577 HG03585 HG03593 HG03594 HG03600 HG03603 HG03615 HG03624 HG03629 HG03636 HG03644 HG03646 HG03649 HG03652 HG03660 HG03663 HG03667 HG03672 HG03679 HG03680 HG03681 HG03685 HG03686 HG03687 HG03691 HG03693 HG03694 HG03695 HG03696 HG03697 HG03702 HG03705 HG03708 HG03711 HG03713 HG03716 HG03718 HG03720 HG03727 HG03729 HG03738 HG03740 HG03742 HG03743 HG03744 HG03745 HG03746 HG03750 HG03753 HG03755 HG03767 HG03771 HG03773 HG03775 HG03777 HG03778 HG03779 HG03785 HG03786 HG03788 HG03790 HG03792 HG03800 HG03803 HG03809 HG03812 HG03815 HG03821 HG03824 HG03830 HG03833 HG03837 HG03844 HG03846 HG03848 HG03850 HG03851 HG03854 HG03856 HG03864 HG03866 HG03867 HG03869 HG03870 HG03871 HG03872 HG03875 HG03885 HG03887 HG03890 HG03896 HG03899 HG03900 HG03902 HG03905 HG03908 HG03911 HG03914 HG03917 HG03920 HG03926 HG03941 HG03943 HG03950 HG03953 HG03960 HG03963 HG03965 HG03967 HG03969 HG03971 HG03974 HG03976 HG03978 HG03985 HG03990 HG03991 HG03998 HG03999 HG04002 HG04003 HG04006 HG04015 HG04017 HG04019 HG04020 HG04022 HG04023 HG04033 HG04039 HG04056 HG04060 HG04061 HG04080 HG04093 HG04094 HG04096 HG04098 HG04100 HG04107 HG04131 HG04134 HG04140 HG04146 HG04152 HG04155 HG04158 HG04161 HG04164 HG04173 HG04176 HG04182 HG04185 HG04188 HG04194 HG04198 HG04206 HG04210 HG04211 HG04219 HG04222 HG04225 HG04229 HG04235 HG04238 HG04239 NA06984 NA06986 NA06994 NA07048 NA07051 NA07347 NA07357 NA10851 NA11829 NA11831 NA11843 NA11881 NA11893 NA11919 NA11930 NA11932 NA11992 NA11994 NA12003 NA12005 NA12043 NA12045 NA12144 NA12154 NA12155 NA12272 NA12282 NA12286 NA12340 NA12342 NA12347 NA12399 NA12413 NA12546 NA12716 NA12748 NA12750 NA12760 NA12762 NA12775 NA12777 NA12812 NA12814 NA12827 NA12829 NA12842 NA12872 NA12874 NA12889 NA18486 NA18498 NA18501 NA18504 NA18507 NA18510 NA18516 NA18519 NA18522 NA18530 NA18534 NA18536 NA18543 NA18544 NA18546 NA18548 NA18549 NA18557 NA18558 NA18559 NA18561 NA18562 NA18563 NA18572 NA18603 NA18605 NA18606 NA18608 NA18609 NA18611 NA18612 NA18613 NA18620 NA18621 NA18622 NA18623 NA18624 NA18629 NA18632 NA18633 NA18635 NA18636 NA18637 NA18638 NA18639 NA18643 NA18645 NA18647 NA18648 NA18740 NA18745 NA18747 NA18748 NA18749 NA18757 NA18853 NA18856 NA18865 NA18868 NA18871 NA18874 NA18877 NA18879 NA18908 NA18910 NA18915 NA18917 NA18923 NA18934 NA18940 NA18943 NA18944 NA18945 NA18948 NA18952 NA18953 NA18959 NA18960 NA18961 NA18962 NA18965 NA18966 NA18967 NA18970 NA18971 NA18974 NA18977 NA18982 NA18983 NA18984 NA18985 NA18986 NA18988 NA18989 NA18990 NA18994 NA18995 NA19000 NA19004 NA19005 NA19006 NA19007 NA19009 NA19012 NA19020 NA19025 NA19026 NA19027 NA19028 NA19031 NA19035 NA19041 NA19043 NA19055 NA19056 NA19058 NA19060 NA19062 NA19063 NA19066 NA19067 NA19068 NA19070 NA19072 NA19075 NA19076 NA19079 NA19082 NA19083 NA19085 NA19086 NA19088 NA19089 NA19091 NA19092 NA19096 NA19098 NA19107 NA19113 NA19117 NA19119 NA19121 NA19130 NA19138 NA19141 NA19144 NA19146 NA19153 NA19160 NA19171 NA19175 NA19184 NA19189 NA19198 NA19200 NA19207 NA19210 NA19213 NA19223 NA19236 NA19239 NA19248 NA19256 NA19307 NA19308 NA19309 NA19312 NA19317 NA19318 NA19319 NA19331 NA19334 NA19346 NA19347 NA19350 NA19360 NA19372 NA19374 NA19375 NA19376 NA19380 NA19383 NA19384 NA19385 NA19393 NA19394 NA19397 NA19428 NA19429 NA19430 NA19443 NA19448 NA19451 NA19452 NA19454 NA19455 NA19461 NA19466 NA19649 NA19652 NA19655 NA19658 NA19661 NA19664 NA19670 NA19676 NA19679 NA19682 NA19700 NA19703 NA19711 NA19717 NA19720 NA19723 NA19726 NA19729 NA19732 NA19735 NA19741 NA19747 NA19750 NA19756 NA19759 NA19762 NA19771 NA19774 NA19777 NA19780 NA19783 NA19786 NA19789 NA19792 NA19795 NA19818 NA19834 NA19900 NA19904 NA19908 NA19916 NA19920 NA19922 NA19982 NA19984 NA20126 NA20278 NA20281 NA20291 NA20298 NA20318 NA20340 NA20342 NA20346 NA20348 NA20351 NA20356 NA20362 NA20509 NA20510 NA20511 NA20512 NA20513 NA20515 NA20516 NA20518 NA20519 NA20520 NA20521 NA20524 NA20525 NA20527 NA20528 NA20532 NA20534 NA20536 NA20538 NA20539 NA20543 NA20544 NA20581 NA20586 NA20588 NA20752 NA20754 NA20755 NA20758 NA20759 NA20762 NA20763 NA20765 NA20767 NA20770 NA20778 NA20783 NA20785 NA20787 NA20792 NA20796 NA20798 NA20801 NA20803 NA20805 NA20806 NA20809 NA20810 NA20811 NA20812 NA20814 NA20815 NA20827 NA20845 NA20846 NA20850 NA20852 NA20858 NA20861 NA20863 NA20864 NA20866 NA20867 NA20870 NA20884 NA20885 NA20887 NA20889 NA20890 NA20891 NA20895 NA20897 NA20901 NA20903 NA20904 NA20905 NA20911 NA21087 NA21090 NA21091 NA21092 NA21093 NA21094 NA21095 NA21098 NA21099 NA21100 NA21104 NA21105 NA21107 NA21109 NA21111 NA21112 NA21113 NA21114 NA21115 NA21116 NA21117 NA21118 NA21119 NA21123 NA21124 NA21126 NA21127 NA21128 NA21129 NA21130 NA21133 NA21135</code></pre>
<p>The last header line describes the format for all of the following lines in the file, each of which represents a position in the genome that has multiple possible alleles. The first eight columns contain information about the position and name of the variant, while all of the remaining columns represent the genotype for a single individual, with their sample ID functioning as the column name in the last header line. You can use the header line definitions to determine what entries in the <code>FILTER</code>, <code>INFO</code>, and genotype fields mean.</p>
</div>
<div id="section-plink" class="section level3">
<h3>PLINK</h3>
<p>In two weeks we’ll be discussing how to use the software tool <a href="https://www.cog-genomics.org/plink/1.9/">PLINK</a> to analyze genotype data, including creating our own PLINK format files. If you are overly anxious to learn about the standard file types, there is good documentation of the <a href="https://www.cog-genomics.org/plink/1.9/input">PLINK input files on the PLINK website</a>.</p>
</div>
</div>
<div id="section-representing-epigenetic-sequencing-data" class="section level2">
<h2>Representing epigenetic sequencing data</h2>
<p>Epigenetics can refer to many different mechanisms of information transmission that is independent of changes to nucleotides. The strict definition includes only heritable phenotype changes, but it is often used to describe functional changes to the genome regardless of whether they are inherited. This can include gene expression, alternative splicing of genes, DNA methylation, chromatin modifications, or histone modifications. When we talk about modifications, such as acetylation, methylation, ubiquitylation, phosphorylation, or sumoylation, we are referring to the addition of a protein or protein group to the chromatin, histone or DNA. The name of this protein or protein group precedes the -ylation term.</p>
<p>Some of these epigenetic modifications can be studied with modified sequencing techniques. For example, DNA methylation is the addition of a methyl group to cytosine. Bisulfite treatment will change a cytosine to uracil (an RNA nucleotide), but will leave 5-methylcytosine (cytosine with a methyl group on the 5’ end) unaffected. This means that after DNA is treated, it can be sequenced to determine which nucleotides were methylated and which were not.</p>
<p>Other epigenetic modifications are studied by a combination of sequencing and other techniques. For example, ChIP-seq is chromatin immunoprecipitation followed by sequencing. This method gives us information about where specific transcription factors are binding. ChIP-seq refers to a technique: knowing that data is coming from ChIP-seq doesn’t tell you what epigenetic data is being obtained in the way that bisulfite sequencing does. You can immunoprecipitate many different proteins, so it’s important to note which one was targeted. ChIP-seq aims to only sequence portions of the genome that were proximate to the targeted protein.</p>
<p>RNA-sequencing is a very well-known source of epigenetic data, likely because it is relatively easy to obtain and data analysis is fairly straightforward to interpret.</p>
<div id="section-how-is-epigenetic-data-different-from-genetic-data" class="section level3">
<h3>How is epigenetic data different from genetic data?</h3>
<p>If we are using sequencing technology, we use the same file types to encode epigenetic data as we would with genetic data. However, our assumptions are different. Multiple reads of the same gene in DNA sequencing means higher coverage. With RNA sequencing, it means more copies of the transcript. With bisulfite sequencing, we care about what proportion of the reads at any particular site have a uracil instead of a cytosine.</p>
</div>
</div>
<div id="section-accessing-data-from-public-databases" class="section level2">
<h2>Accessing data from public databases</h2>
<p>If you end up working as a bioinformatician or computational geneticist, you will likely spend a large amount of time gaining access to other people’s datasets in order to replicate your results or gain additional power for statistical analyses. Here are some good places to start looking:</p>
<div id="section-open-access-databases" class="section level3">
<h3>Open access databases</h3>
<ul>
<li><a href="https://portal.gdc.cancer.gov/">TCGA: The Cancer Genome Atlas</a></li>
<li><a href="https://www.internationalgenome.org/data/">1000 Genomes</a></li>
<li><a href="https://www.sanger.ac.uk/resources/downloads/human/hapmap3.html">HapMap3</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/guide/all/">NCBI has a variety of databases available</a>, some of which include genetic data</li>
<li><a href="https://www.ncbi.nlm.nih.gov/geo/">GEO: Gene Expression Omnibus</a></li>
</ul>
</div>
<div id="section-restricted-access-databases" class="section level3">
<h3>Restricted access databases</h3>
<p>Because genetic data can be used to identify individuals, most genetic data is restricted access, especially if it is associated with any phenotypes. Many of these databases are open to anybody with a good question and university credentials, so consider using restricted access data as an opportunity to clarify your ideas before getting too deep into any analyses.</p>

<script type="application/shiny-prerendered" data-context="server-start">
library(learnr)
library(dplyr)
library(ggplot2)
library(gtools)
library(scales)

knitr::opts_chunk$set(echo = FALSE, comment=NA, error=TRUE)
knitr::opts_knit$set(root.dir = "week_1_GeneticData")


local({
  hook_output <- knitr::knit_hooks$get('output')
  knitr::knit_hooks$set(output = function(x, options) {
    if (!is.null(options$max.height)) options$attr.output <- c(
      options$attr.output,
      sprintf('style="max-height: %s;"', options$max.height)
    )
    hook_output(x, options)
  })
})
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::register_http_handlers(session, metadata = NULL)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::clear_exercise_setup_chunks()
</script>
 
<script type="application/shiny-prerendered" data-context="server">
session$onSessionEnded(function() {
        learnr:::event_trigger(session, "session_stop")
      })
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_setup_chunk("__setup__", c("library(learnr)", "library(dplyr)", "library(ggplot2)", "library(gtools)", 
"library(scales)", "", "knitr::opts_chunk$set(echo = FALSE, comment=NA, error=TRUE)", 
"knitr::opts_knit$set(root.dir = \"week_1_GeneticData\")", "", 
"", "local({", "  hook_output <- knitr::knit_hooks$get('output')", 
"  knitr::knit_hooks$set(output = function(x, options) {", "    if (!is.null(options$max.height)) options$attr.output <- c(", 
"      options$attr.output,", "      sprintf('style=\"max-height: %s;\"', options$max.height)", 
"    )", "    hook_output(x, options)", "  })", "})"), overwrite = FALSE)
</script>
 
<script type="application/shiny-prerendered" data-context="server">
learnr:::store_exercise_setup_chunk("__setup__", c("library(learnr)", "library(dplyr)", "library(ggplot2)", "library(gtools)", 
"library(scales)", "", "knitr::opts_chunk$set(echo = FALSE, comment=NA, error=TRUE)", 
"knitr::opts_knit$set(root.dir = \"week_1_GeneticData\")", "", 
"", "local({", "  hook_output <- knitr::knit_hooks$get('output')", 
"  knitr::knit_hooks$set(output = function(x, options) {", "    if (!is.null(options$max.height)) options$attr.output <- c(", 
"      options$attr.output,", "      sprintf('style=\"max-height: %s;\"', options$max.height)", 
"    )", "    hook_output(x, options)", "  })", "})"), overwrite = FALSE)
</script>
 <!--html_preserve-->
<script type="application/shiny-prerendered" data-context="dependencies">
{"type":"list","attributes":{},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootstrap"]},{"type":"character","attributes":{},"value":["3.3.5"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/bootstrap"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["viewport"]}},"value":[{"type":"character","attributes":{},"value":["width=device-width, initial-scale=1"]}]},{"type":"character","attributes":{},"value":["js/bootstrap.min.js","shim/html5shiv.min.js","shim/respond.min.js"]},{"type":"character","attributes":{},"value":["css/cerulean.min.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["pagedtable"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/pagedtable-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["js/pagedtable.js"]},{"type":"character","attributes":{},"value":["css/pagedtable.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["textmate.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.10.1.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.1.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-autocompletion"]},{"type":"character","attributes":{},"value":["0.10.1.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-autocompletion.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.1.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-diagnostics"]},{"type":"character","attributes":{},"value":["0.10.1.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-diagnostics.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.1.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-format"]},{"type":"character","attributes":{},"value":["0.10.1.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmarkdown/templates/tutorial/resources"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-format.js"]},{"type":"character","attributes":{},"value":["tutorial-format.css","rstudio-theme.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.1.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["navigation"]},{"type":"character","attributes":{},"value":["1.1"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/navigation-1.1"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tabsets.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["highlightjs"]},{"type":"character","attributes":{},"value":["9.12.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/highlightjs"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["highlight.js"]},{"type":"character","attributes":{},"value":["default.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["jquery"]},{"type":"character","attributes":{},"value":["1.11.3"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/jquery"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["jquery.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["font-awesome"]},{"type":"character","attributes":{},"value":["5.1.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["rmd/h/fontawesome"]}]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["css/all.css","css/v4-shims.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["rmarkdown"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["2.4"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["bootbox"]},{"type":"character","attributes":{},"value":["4.4.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/bootbox"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["bootbox.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.1.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["idb-keyvalue"]},{"type":"character","attributes":{},"value":["3.2.0"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/idb-keyval"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["idb-keyval-iife-compat.min.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[false]},{"type":"character","attributes":{},"value":["0.10.1.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial"]},{"type":"character","attributes":{},"value":["0.10.1.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial.js"]},{"type":"character","attributes":{},"value":["tutorial.css"]},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.1.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-autocompletion"]},{"type":"character","attributes":{},"value":["0.10.1.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-autocompletion.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.1.9006"]}]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["name","version","src","meta","script","stylesheet","head","attachment","package","all_files","pkgVersion"]},"class":{"type":"character","attributes":{},"value":["html_dependency"]}},"value":[{"type":"character","attributes":{},"value":["tutorial-diagnostics"]},{"type":"character","attributes":{},"value":["0.10.1.9006"]},{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["file"]}},"value":[{"type":"character","attributes":{},"value":["lib/tutorial"]}]},{"type":"NULL"},{"type":"character","attributes":{},"value":["tutorial-diagnostics.js"]},{"type":"NULL"},{"type":"NULL"},{"type":"NULL"},{"type":"character","attributes":{},"value":["learnr"]},{"type":"logical","attributes":{},"value":[true]},{"type":"character","attributes":{},"value":["0.10.1.9006"]}]}]}
</script>
<!--/html_preserve-->
<!--html_preserve-->
<script type="application/shiny-prerendered" data-context="execution_dependencies">
{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages"]}},"value":[{"type":"list","attributes":{"names":{"type":"character","attributes":{},"value":["packages","version"]},"class":{"type":"character","attributes":{},"value":["data.frame"]},"row.names":{"type":"integer","attributes":{},"value":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58]}},"value":[{"type":"character","attributes":{},"value":["backports","base","colorspace","compiler","crayon","curl","datasets","digest","dplyr","ellipsis","evaluate","farver","fastmap","generics","ggplot2","glue","graphics","grDevices","grid","gtable","gtools","htmltools","htmlwidgets","httpuv","jsonlite","knitr","labeling","later","learnr","lifecycle","magrittr","markdown","methods","mime","munsell","pillar","pkgconfig","promises","purrr","R6","Rcpp","rlang","rmarkdown","rprojroot","scales","shiny","stats","stringi","stringr","tibble","tidyselect","tools","utils","vctrs","withr","xfun","xtable","yaml"]},{"type":"character","attributes":{},"value":["1.1.10","4.0.2","1.4-1","4.0.2","1.3.4","4.3","4.0.2","0.6.25","1.0.2","0.3.1","0.14","2.0.3","1.0.1","0.0.2","3.3.2","1.4.2","4.0.2","4.0.2","4.0.2","0.3.0","3.8.2","0.5.0.9001","1.5.1","1.5.4","1.7.1","1.30","0.3","1.1.0.1","0.10.1.9006","0.2.0","1.5","1.1","4.0.2","0.9","0.5.0","1.4.6","2.0.3","1.1.1","0.3.4","2.4.1","1.0.5","0.4.7","2.4","1.3-2","1.1.1","1.5.0","4.0.2","1.5.3","1.4.0","3.0.3","1.1.0","4.0.2","4.0.2","0.3.4","2.3.0","0.18","1.8-4","2.2.1"]}]}]}
</script>
<!--/html_preserve-->
</div>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="section-fn1"><p>FASTA was developed as the input file for a program of the same name that searched for sequence similarities between DNA, mRNA, or protein sequences. The modern analog of this tools is <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi">BLAST</a>, which still uses the FASTA file format as its input.<a href="#section-fnref1" class="footnote-back">↩︎</a></p></li>
<li id="section-fn2"><p>You can see a full description of nucleotide and amino acid encodings used by BLAST <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs&amp;DOC_TYPE=BlastHelp">here</a>. The International Union of Biochemistry (IUB) and International Union of Pure and Applied Chemistry (IUPAC) standardized representation of common biochemical from 1959-1985.All but the first five IUB/IUPAC codes represent <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC322779/pdf/pnas00305-0018.pdf">incomplete bases</a>, where there could be multiple nucleotide bases. These ambiguity characters were designed to encode positional variations in order to report DNA sequencing errors, consensus sequences, or single-nucleotide polymorphisms.<a href="#section-fnref2" class="footnote-back">↩︎</a></p></li>
<li id="section-fn3"><p><a href="https://en.wikipedia.org/wiki/Uracil">Uracil</a> is <a href="https://en.wikipedia.org/wiki/Deamination#Cytosine">thymine without a methyl group</a> and replaces thymine in RNA, bonding with adenine. When the International Union of Biochemistry (IUB) and International Union of Pure and Applied Chemistry (IUPAC) codes were being <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC322779/pdf/pnas00305-0018.pdf">rewritten in the 1980s</a>, the committee decided that RNA nitrogenous bases were best represented by their DNA counterparts. For this reason, the FASTA files for RNA sequences on NCBI are encoded as their reverse complements (the cDNA). However, you <em>can</em> include uracil (U) in your FASTA files as it was a part of the <a href="https://www.jbc.org/content/241/3/527.full.pdf">original 1965 recommendations</a> and BLAST will understand what it means.<a href="#section-fnref3" class="footnote-back">↩︎</a></p></li>
<li id="section-fn4"><p>Cytosine, thymine, and uracil have only one nitrogen ring in their structures, making them <a href="https://en.wikipedia.org/wiki/Pyrimidine">pyrimidines</a>.<a href="#section-fnref4" class="footnote-back">↩︎</a></p></li>
<li id="section-fn5"><p>Adenine and guanine have two nitrogen rings in their structures, making them <a href="https://en.wikipedia.org/wiki/Purine">purines</a>.<a href="#section-fnref5" class="footnote-back">↩︎</a></p></li>
<li id="section-fn6"><p>Because of their structures, cytosine and guanine form <a href="https://en.wikipedia.org/wiki/Base_pair">three hydrogen bonds</a> with each other, which is a stronger interaction than that between adenine and thymine or uracil. This strong interaction is the reason that DNA with <a href="https://en.wikipedia.org/wiki/GC-content">high GC-content</a> is more stable.<a href="#section-fnref6" class="footnote-back">↩︎</a></p></li>
<li id="section-fn7"><p>Adenine and thymine or uracil form only two hydrogen bonds with each other.<a href="#section-fnref7" class="footnote-back">↩︎</a></p></li>
<li id="section-fn8"><p>Guanine, thymine, and uracil are <a href="https://en.wikipedia.org/wiki/Ketone#Cyclic_ketones">ketones</a>, which means they have at least one carbon-oxygen double bond (<a href="https://en.wikipedia.org/wiki/Carbonyl_group">carbonyl group</a>). Guanine has one carbonyl group and thymine and uracil have two. While cytosine also has a carbonyl group, this group is the <a href="https://en.wikipedia.org/wiki/Cytosine#Chemical_reactions">hydrogen bond acceptor</a> rather than the donor.<a href="#section-fnref8" class="footnote-back">↩︎</a></p></li>
<li id="section-fn9"><p>Adenine and cytosine have an <a href="https://en.wikipedia.org/wiki/Amine">amino group (NH^2)</a> that hydrogen bonds to the carbonyl group of the ketone bases. Guanine also has an amino group, but - like with cytosine’s ketone group - this amine is the <a href="https://en.wikipedia.org/wiki/Guanine#Properties">hydrogen bond acceptor</a>, not the donor.<a href="#section-fnref9" class="footnote-back">↩︎</a></p></li>
<li id="section-fn10"><p>While you won’t usually see the “-” character in FASTA files, it is useful for alignments or representing unknown bases.<a href="#section-fnref10" class="footnote-back">↩︎</a></p></li>
<li id="section-fn11"><p>As mentioned before, NCBI prefers to represent RNA as DNA, so this sequence is the reverse complement of the mRNA.<a href="#section-fnref11" class="footnote-back">↩︎</a></p></li>
<li id="section-fn12"><p>Note that this file is <a href="https://ftp-trace.ncbi.nih.gov/1000genomes/ftp/phase3/data/NA19238/sequence_read/">309 MB</a> compressed, so the download may take some time. I would also strongly suggest that you don’t unzip it.<a href="#section-fnref12" class="footnote-back">↩︎</a></p></li>
<li id="section-fn13"><p><code>gzcat</code> allows you to view the contents of a gzipped file without unzipping it.<a href="#section-fnref13" class="footnote-back">↩︎</a></p></li>
<li id="section-fn14"><p>I learned while preparing this document that you can also repeat the sequence ID after the “+” like in the header line (“+ERR000018.1 BGI-FC307N0AAXX_5_1_390:1819”), but I have never personally encountered this notation.<a href="#section-fnref14" class="footnote-back">↩︎</a></p></li>
</ol>
</div>

</div> <!-- topics -->

<div class="topicsContainer">
<div class="topicsPositioner">
<div class="band">
<div class="bandContent topicsListContainer">

<!-- begin doc-metadata -->
<div id="doc-metadata">
<h2 class="title toc-ignore" style="display:none;">Genetic Data Tutorial</h2>
</div>
<!-- end doc-metadata -->

</div> <!-- bandContent.topicsListContainer -->
</div> <!-- band -->
</div> <!-- topicsPositioner -->
</div> <!-- topicsContainer -->


</div> <!-- bandContent page -->
</div> <!-- pageContent band -->




<script>
// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});
</script>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>


</body>

</html>
